go/vt/vttablet/tabletmanager/vreplication/engine.go

```diff
@@ -393,7 +394,18 @@ func (vre *Engine) exec(query string, runAsAdmin bool) (*sqltypes.Result, error)
 		}
 
 		vdbc := newVDBClient(dbClient, binlogplayer.NewStats())
-		for id := int32(qr.InsertID); id < int32(maxInsert); id++ {
+
+		// If we are creating multiple streams, for example in a merge workflow going from 2 shards to 1 shard,
+		// we will be inserting multiple rows. To get the ids of subsequent streams we need to know what
+		// the auto_increment_increment step is. In a multi-master environment, for example, we will encounter
+		// auto increment steps > 1.
+		autoIncrementStep, err := vre.getAutoIncrementStep(dbClient)
+		if err != nil {
+			return nil, err
+		}
+		firstID := int32(qr.InsertID)
```

Comment:
Confirmed that this is indeed an int32: https://github.com/vitessio/vitess/blob/main/go/vt/sidecardb/schema/vreplication/vreplication.sql#L19